docs/cookbook/production-deployment.md

```diff
@@ -0,0 +1,308 @@
+ln -s /var/lib/docker/volumes/production_web-volume/_data/ web
+```
+and start containers with docker-compose.
+```
+docker-compose -p production up -d
+docker-compose -p production exec php-fpm ./phing build-new
+```
+
+Now the docker container should run.
+We want to setup scheduler for execution of cron jobs by adding one line into `/etc/crontab` file.
+Cron job is executed every 5 minutes in `php-fpm` container.
+```
+*/5 * * * * /usr/bin/docker exec production-php-fpm php phing cron
+```
+Since web application is running we can go to administration and change passwords for default administrators.
+With login `superadmin` and password `admin123` we can do it via these urls:
+* `/admin/administrator/edit/1`
+* `/admin/administrator/edit/2`
+
+Now we go to `/admin/dashboard/` and fulfill all requests that are demanding for us by red colored links.
+
+#### Next Deploys
+
+To preserve created data we need to use phing target `build` that consists of two commands 
+* `build-deploy-part-1-db-independent` no maintenance page is needed during execution of this command 
+* `build-deploy-part-2-db-dependent` maintenance page is needed if there are exist unapplied database migrations  
+
+With each update of master branch in our repository we need to rebuild image based on [Docker Image Building](#docker_image_building) section.
+
+We are logged in production server via ssh and start to deploy new built production image.
+```
+cd PATH_TO_DIRECTORY_WHERE_IS_DOCKER_COMPOSE_YML_STORED (e.g. /var/www/html)
+
+# make sure that container for building the application image is cleared
+docker rm -f build-php-fpm-container
+
+# launch container for building the application image
+docker run --detach --rm --name build-php-fpm-container \
+    --add-host redis:192.168.0.1 --add-host postgres:192.168.0.1 --add-host elasticsearch:192.168.0.1 --add-host smpt-server:192.168.0.1 \
+    --network production_shopsys-network \
+    production-php-fpm \
+    docker-php-entrypoint php-fpm
+
+# launch build part1 that doesn't have impact on actual running production container
+docker exec build-php-fpm-container php phing build-deploy-part-1-db-independent
+
+# turn on maintenance page on actual running production container
+docker-compose -p production exec php-fpm php phing maintenance-on
+
+# launch build part2 that could have impact on actual running production container
+docker exec build-php-fpm-container php phing build-deploy-part-2-db-dependent
+
+# save the changes of container into the image production-php-fpm
+docker commit build-php-fpm-container production-php-fpm
+
+# remove actual running production container with the web volume that is shared with webserver container
+docker-compose -p production rm -fs php-fpm
+docker volume rm -f production_web-volume
+
+# recreate production container with the reload of webserver container with new web folder mounted
+docker-compose -p production up -d --force-recreate webserver
```

Comment:
Without deleting the php-fpm container, wouldnt this just recreate it with newer version, is there need to delete them before it, I think that docker-compose can handle it by himself.